File: frontend/lib/api/ide-session.ts
import apiClient from '../api-client';

export interface IdeFile {
  path: string;
  content: string;
  language: string;
}

export interface CursorPosition {
  line: number;
  column: number;
}

export interface IdeSessionState {
  openFiles: IdeFile[];
  activeFile: string | null;
  cursorPositions: Record<string, CursorPosition>;
}

export interface IdeSessionResponse {
  sessionId: string;
  userId: string;
  sessionState: IdeSessionState;
  lastSavedAt: string;
}

/**
 * Save IDE session state to backend
 */
export async function saveIdeSession(userId: string, sessionState: IdeSessionState): Promise<void> {
  await apiClient.post('/api/ide/session', {
    userId,
    sessionState,
  });
}

/**
 * Load IDE session state from backend
 */
export async function loadIdeSession(userId: string): Promise<IdeSessionResponse | null> {
  try {
    const response = await apiClient.get<IdeSessionResponse>(`/api/ide/session/${userId}`);
    return response.data;
  } catch (error: any) {
    if (error.response?.status === 404) {
      return null; // No session found
    }
    throw error;
  }
}

/**
 * Delete IDE session
 */
export async function deleteIdeSession(userId: string): Promise<void> {
  await apiClient.delete(`/api/ide/session/${userId}`);
}

File: src/Shared/Models/IdeSession.cs
namespace Shared.Models;

/// <summary>
/// Represents an IDE session state for a user
/// </summary>
public class IdeSession : BaseEntity
{
    public string UserId { get; set; } = string.Empty;
    public string SessionData { get; set; } = string.Empty; // JSON serialized session state
    public DateTime LastSavedAt { get; set; }
}

/// <summary>
/// DTO for IDE session state
/// </summary>
public class IdeSessionState
{
    public List<IdeFile> OpenFiles { get; set; } = new();
    public string? ActiveFile { get; set; }
    public Dictionary<string, CursorPosition> CursorPositions { get; set; } = new();
}

/// <summary>
/// Represents a file in the IDE
/// </summary>
public class IdeFile
{
    public string Path { get; set; } = string.Empty;
    public string Content { get; set; } = string.Empty;
    public string Language { get; set; } = "csharp";
}

/// <summary>
/// Represents cursor position in a file
/// </summary>
public class CursorPosition
{
    public int Line { get; set; }
    public int Column { get; set; }
}
